### PR TITLE
Add image build artifacts and code for the configmap-puller sidecar container for traefik

### PR DESCRIPTION
**Pull request by maulikjs**

### CODE REVIEW

File: configmap-puller/cmd/configmap-puller/main.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
+// Note: the example only works with the code within the same release/branch.
```
Comment (erdii) [inline configmap-puller/cmd/configmap-puller/main.go]: ups - a copy/paste artifact i forgot to remove! :sweat_smile:

File: configmap-puller/cmd/configmap-puller/main.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	"k8s.io/client-go/tools/clientcmd"
+	//
+	// Uncomment to load all auth plugins
+	// _ "k8s.io/client-go/plugin/pkg/client/auth"
+	//
+	// Or uncomment to load specific auth plugins
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/azure"
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/openstack"
```
Comment (erdii) [inline configmap-puller/cmd/configmap-puller/main.go]: copy-paste artifact as well

**Review by erdii — CHANGES_REQUESTED** [review]
found some forgotten lines of code. can you please remove them?

File: configmap-puller/cmd/configmap-puller/main.go (hunk 1 above)
Comment (maulikjs) [reply configmap-puller/cmd/configmap-puller/main.go]: oops missed out on that one! Done

**Review by tmckayus — APPROVED** [review]

**Comment by maulikjs** [comment]
As per request
* why it is in ODH
As part of HA-izing Jupyterhub, we are using this as a sidecar container that would update the rules.toml file read by the traefik proxy whenever the configmap `traefik-rules` is updated by Jupyterhub.
It is deployed as a part of odh and therefore should be included in ODH images.

* steps to build the image
```
cd configmap-puller
DOCKER_IMAGE=quay.io/opendatahub/configmap-puller:0.1 make build
```

**Comment by vpavlin** [comment]
I am good with the code, but I did not see a README - can you add a README with the info from you comment to the PR?

**Review by vpavlin — APPROVED** [review]

**Pull request merged**